update.sh

```diff
@@ -1,16 +1,18 @@
 #!/bin/bash
-# 1st param (optional): target branch. Defaults to master.
-TARGET_BRANCH=${1:-master}
+# 1st param (optional): force the target branch to update to. WILL LOSE UNCOMMITTED CHANGES. Defaults to staying on current branch.
 
 set -ex
 
 cd `dirname $0`
 
-git checkout $TARGET_BRANCH
-git pull origin $TARGET_BRANCH
+if [[ -n $1 ]]
+then
+	git fetch origin
+	git checkout --force origin/$1
+fi
 
 git submodule sync
 git submodule update --init --recursive
 
-git submodule foreach python setup.py develop
+git submodule foreach python setup.py develop --user
```

Comment:
This doesn't work if you are using a virtualenv, but it's ok for production servers.